src/main/java/co/elastic/logstash/filters/elasticintegration/geoip/ManagedIpDatabaseHolder.java

```diff
@@ -51,7 +50,7 @@ public String getTypeIdentifier() {
 
     @Override
     public String info() {
-        return withLock(readLock, () -> String.format("ManagedGeoipDatabase{type=%s, valid=%s}", getTypeIdentifier(), isValid()));
+        return withLock(readLock, () -> String.format("ManagedIpDatabase{type=%s, valid=%s}", getTypeIdentifier(), isValid()));
     }
 
     public void setDatabasePath(final String newDatabasePath) {
```

Comment:
When a managed database has its database path updated, we need to close the reader to the previous one, something like:

~~~ java
    public void setDatabasePath(final String newDatabasePath) {
        final IpDatabaseAdapter previousDatabase = withLock(writeLock, () -> {
            final IpDatabaseAdapter localPreviousDatabase = this.currentDatabase;
            this.currentDatabase = Optional.ofNullable(newDatabasePath)
                    .map(Paths::get)
                    .map(this::loadDatabase)
                    .orElse(null);
            return localPreviousDatabase;
        });

        if (previousDatabase != null) {
            IOUtils.closeWhileHandlingException(previousDatabase::closeReader);
        }
    }
~~~